dwave/optimization/_model.pyx

```diff
@@ -664,6 +723,26 @@ cdef class _Graph:
             num_edges += self._graph.nodes()[i].get().successors().size()
         return num_edges
 
+    cpdef Py_ssize_t num_inputs(self) noexcept:
```

Comment:
I don't think we test this function currently.

Reply:
Added tests in https://github.com/dwavesystems/dwave-optimization/pull/269/commits/31467f84a760c3f2bf3c37231355f54538b3829f